src/tensorlake/applications/interface/request_context.py

```diff
@@ -42,12 +42,16 @@ def counter(self, name: str, value: int = 1):
 class FunctionProgress:
     """Abstract interface for reporting Tensorlake Function call progress."""
 
-    def update(self, current: float, total: float) -> None:
+    def update(
+        self, current: float, total: float, message: str | None = None, **kwargs
+    ) -> None:
         """Update the progress of the current Tensorlake Function call execution.
 
         Args:
-            current: Current request progress value
-            total: Total request progress value
+            current: Current function call execution step
+            total: Total number of function call execution steps
+            message: Optional message to display with the progress update
+            kwargs: Additional keyword arguments to pass to the progress update
```

Comment:
How are the kwargs are going to be serialized and deserialized? Or they won't be?